packages/meditrak-server/src/routes/importSurveys/ConfigImporter/processCalculatedConfig.js

```diff
@@ -0,0 +1,195 @@
+ * formula: (5ed7558b61f76a6ba5003114 / 5ec5bf8961f76a18ad00d4f9) * 100
+ * @param {*} models
+ * @param {*} formula
+ * @param {*} codes
+ */
+const translateFormula = async (models, formula, codes) => {
+  let translatedFormula = formula;
+
+  for (let i = 0; i < codes.length; i++) {
+    const code = codes[i];
+    const { id: questionId } = await models.question.findOne({ code });
+    translatedFormula = translatedFormula.replace(code, questionId);
+  }
+
+  return translatedFormula;
+};
+
+/**
+ * defaultValues: SchFF00Section1Total=0,SchFF00Section1TotalObtainable=0
+ * will be converted to:
+ * defaultValues:{
+ *     5ed7558b61f76a6ba5003114:0,
+ *     5ec5bf8961f76a18ad00d4f9:0
+ * },
+ * @param {*} models
+ * @param {*} defaultValuesConfig
+ */
+const translateDefaultValues = async (models, defaultValuesConfig) => {
+  const defaultValues = splitStringOnComma(defaultValuesConfig);
+  const translatedDefaultValues = {};
+
+  for (let i = 0; i < defaultValues.length; i++) {
+    const [code, defaultValue] = splitStringOn(defaultValues[i], '=');
+    const { id: questionId } = await models.question.findOne({ code });
+    translatedDefaultValues[questionId] = defaultValue;
+  }
+
+  return translatedDefaultValues;
+};
+
+/**
+ * valueTranslation: SchCVD016a.Yes=3,SchCVD016a.No=0
+ * will be converted to:
+ * valueTranslation: {
+ *		5ed7558b61f76a6ba5003114: { Yes: 3, No: 0 }
+ * }
+ * @param {*} models
+ * @param {*} valueTranslationConfig
+ */
+const translateValueTranslation = async (models, valueTranslationConfig) => {
+  const valueTranslation = splitStringOnComma(valueTranslationConfig);
+  const translatedValueTranslation = {};
+
+  for (let i = 0; i < valueTranslation.length; i++) {
+    const [code, value] = splitStringOn(valueTranslation[i], '=');
+    const [questionCode, questionOption] = splitStringOn(code, '.');
+    const { id: questionId } = await models.question.findOne({ code: questionCode });
+
+    if (!translatedValueTranslation[questionId]) {
+      translatedValueTranslation[questionId] = {};
+    }
+    translatedValueTranslation[questionId][questionOption] = value;
+  }
+
+  return translatedValueTranslation;
+};
+
+/**
+ * text: SchFF00CalcTotal / SchFF00CalcTotalObtainable = $result
+ * Question codes will be replaced by question ids
+ * @param {*} models
+ * @param {*} text
+ * @param {*} codes
+ */
+const translateText = async (models, text, codes) => {
+  let translatedText = text;
+
+  for (let i = 0; i < codes.length; i++) {
+    const code = codes[i];
+    const { id: questionId } = await models.question.findOne({ code });
+    translatedText = translatedText.replace(code, questionId);
+  }
+
+  return translatedText;
+};
+
+const translateConditionConfig = async (models, config) => {
+  return {
+    conditions: await translateConditions(models, config.conditions),
+  };
+};
+
+/**
+ * conditions: Yes.SchCVD016a.operator=GE,Yes.SchCVD016a.operand=5,No.SchCVD016a.operator=LT,No.SchCVD016a.operand=5
```

Comment:
I wonder if we should try to align this with the conditional formula text being developed for the analyticCount indicator @kael89